File: android/app/src/main/java/com/ronaldrobbins/registerdropcounter/Theme.kt
package com.ronaldrobbins.registerdropcounter

import androidx.compose.ui.graphics.Color

object RdcColor {
    var dark: Boolean = false

    val navy: Color get() = if (dark) Color(0xFF102C32) else Color(0xFF1F4E79)
    val navyDeep: Color get() = if (dark) Color(0xFF163E46) else Color(0xFF163A5F)
    val navyMid: Color get() = if (dark) Color(0xFF1E5A64) else Color(0xFF2E75B6)
    val onNavy: Color get() = if (dark) Color(0xFFE7F6F2) else Color(0xFFF7FBFF)

    val navyFg: Color get() = if (dark) Color(0xFF6EE7D4) else Color(0xFF1F4E79)
    val sheet: Color get() = if (dark) Color(0xFF060809) else Color(0xFFEEF1F4)
    val paper: Color get() = if (dark) Color(0xFF10171A) else Color(0xFFFFFFFF)
    val ink: Color get() = if (dark) Color(0xFFE7F6F2) else Color(0xFF1A242E)
    val muted: Color get() = if (dark) Color(0xFF8FB3AE) else Color(0xFF5C6B7A)
    val grid: Color get() = if (dark) Color(0xFF2C454B) else Color(0xFFC5D0DB)
    val input: Color get() = if (dark) Color(0xFFF5C542) else Color(0xFFFFF4C2)
    val computed: Color get() = if (dark) Color(0xFF1A272B) else Color(0xFFF8E4D4)
    val ok: Color get() = if (dark) Color(0xFF0D3F35) else Color(0xFFC6EFCE)
    val okInk: Color get() = if (dark) Color(0xFF6EF0C8) else Color(0xFF006100)
    val bad: Color get() = if (dark) Color(0xFF4D1F28) else Color(0xFFFFC7CE)
    val badInk: Color get() = if (dark) Color(0xFFFFB0BC) else Color(0xFF9C0006)
    val cellInk: Color get() = if (dark) Color(0xFF1C1406) else Color(0xFF1A242E)
    val dropHit: Color get() = if (dark) Color(0xFF15564A) else Color(0xFFD6EADF)
}

File: ios/RegisterDropCounter/Theme.swift
import SwiftUI
import UIKit

enum Theme {
    static let navy = adaptive(
        light: (31 / 255, 78 / 255, 121 / 255),
        dark: (16 / 255, 44 / 255, 50 / 255)
    )
    static let navyDeep = adaptive(
        light: (22 / 255, 58 / 255, 95 / 255),
        dark: (22 / 255, 62 / 255, 70 / 255)
    )
    static let navyMid = adaptive(
        light: (46 / 255, 117 / 255, 182 / 255),
        dark: (30 / 255, 90 / 255, 100 / 255)
    )
    static let navyFg = adaptive(
        light: (31 / 255, 78 / 255, 121 / 255),
        dark: (110 / 255, 231 / 255, 212 / 255)
    )
    static let input = adaptive(
        light: (255 / 255, 244 / 255, 194 / 255),
        dark: (245 / 255, 197 / 255, 66 / 255)
    )
    static let computed = adaptive(
        light: (248 / 255, 228 / 255, 212 / 255),
        dark: (26 / 255, 39 / 255, 43 / 255)
    )
    static let ok = adaptive(
        light: (198 / 255, 239 / 255, 206 / 255),
        dark: (13 / 255, 63 / 255, 53 / 255)
    )
    static let okInk = adaptive(
        light: (20 / 255, 90 / 255, 50 / 255),
        dark: (110 / 255, 240 / 255, 200 / 255)
    )
    static let bad = adaptive(
        light: (255 / 255, 199 / 255, 206 / 255),
        dark: (77 / 255, 31 / 255, 40 / 255)
    )
    static let badInk = adaptive(
        light: (140 / 255, 30 / 255, 40 / 255),
        dark: (255 / 255, 176 / 255, 188 / 255)
    )
    static let cellInk = adaptive(
        light: (26 / 255, 36 / 255, 46 / 255),
        dark: (28 / 255, 20 / 255, 6 / 255)
    )

    static let sheet = adaptive(
        light: (238 / 255, 241 / 255, 244 / 255),
        dark: (6 / 255, 8 / 255, 9 / 255)
    )
    static let paper = adaptive(
        light: (1, 1, 1),
        dark: (16 / 255, 23 / 255, 26 / 255)
    )
    static let ink = adaptive(
        light: (26 / 255, 36 / 255, 46 / 255),
        dark: (231 / 255, 246 / 255, 242 / 255)
    )
    static let muted = adaptive(
        light: (90 / 255, 102 / 255, 114 / 255),
        dark: (143 / 255, 179 / 255, 174 / 255)
    )
    static let grid = adaptive(
        light: (197 / 255, 208 / 255, 219 / 255),
        dark: (44 / 255, 69 / 255, 75 / 255)
    )

    private static func adaptive(
        light: (CGFloat, CGFloat, CGFloat),
        dark: (CGFloat, CGFloat, CGFloat)
    ) -> Color {
        Color(uiColor: UIColor { trait in
            let c = trait.userInterfaceStyle == .dark ? dark : light
            return UIColor(red: c.0, green: c.1, blue: c.2, alpha: 1)
        })
    }
}